assets/js/googlesitekit/datastore/user/key-metrics.js

```diff
@@ -112,30 +129,70 @@ const baseReducer = ( state, { type, payload } ) => {
 	}
 };
 
-const baseResolvers = {};
+const baseResolvers = {
+	*getUserPickedMetrics() {
+		const registry = yield Data.commonActions.getRegistry();
+		const userPickedKeyMetrics = registry
+			.select( CORE_USER )
+			.getUserPickedMetrics();
+
+		if ( userPickedKeyMetrics ) {
+			return userPickedKeyMetrics;
+		}
+
+		yield fetchGetUserPickedMetricsStore.actions.fetchGetUserPickedMetrics();
+	},
+};
 
 const baseSelectors = {
+	/**
+	 * Gets key metrics for this user, either from the user-selected
+	 * key metrics selected by this user (if available) or—if the user has not
+	 * manually selected their own key metrics—from the automatically-selected
+	 * (eg. "answer-based") metrics based on their answers to our User Input
+	 * questions.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Object|undefined)} Key metrics settings.
+	 */
+	getKeyMetrics: createRegistrySelector( ( select ) => () => {
+		const userPickedMetrics = select( CORE_USER ).getUserPickedMetrics();
+
```

Comment:
Hmm, it seems to me we should do one of two things here - either, `await` for `getUserPickedMetrics()` to be resolved via `__experimentalResolveSelect()`, _or_ check if `userPickedKeyMetrics` is `undefined` and return `undefined` if so.

Otherwise, if `getAnswerBasedMetrics()` has already been called and resolved prior to `getKeyMetrics()`, `getKeyMetrics()` will return the value of `getAnswerBasedMetrics()` the first time it's called, even if `getUserPickedMetrics()` is going to return something.

Reply:
Good call/catch!

I've gone with the `undefined` route as it's easier (just inside a selector) and the approach we'd usually use.

I'll add some notes about a QA:Eng brief here as well. 👍🏻